integrations/llama_cpp/src/haystack_integrations/components/generators/llama_cpp/chat/chat_generator.py

```diff
@@ -21,6 +21,10 @@ def _convert_message_to_llamacpp_format(message: ChatMessage) -> Dict[str, str]:
     if message.name:
         formatted_msg["name"] = message.name
 
+    if formatted_msg["role"] == "tool":
+        formatted_msg["name"] = message.tool_call_result.origin.tool_name
+        formatted_msg["content"] = message.tool_call_result.result
```

Comment:
starting from 2.9.0, Haystack `function` messages will be automatically converted to `tool` messages. This change ensures compatibility.